lighthouse-core/report/html/renderer/details-renderer.js

```diff
@@ -264,6 +272,22 @@ class DetailsRenderer {
     return tableElem;
   }
 
+  /**
+   * @param {TableDetailsJSON} details
+   * @return {Element}
+   */
+  _renderList(details) {
+    const listContainer = this._dom.createElement('div', 'lh-list');
+
+    details.items.forEach(item => {
+      const itemEl = this._dom.createElement('div');
+      itemEl.append(this.render(item));
```

Comment:
I feel like we had to avoid forward looking nested details types because of protobuf/LR concerns, but I don't remember exactly where that landed.

was this a real limitation @exterkamp or am I making that up?

if we do, maybe we just keep it a `code-snippet-list` for now

Reply:
I don't know what a forward looking nested details type is, but would this also affect the `DetailsRendererNodeDetailsJSON` in `DetailsRendererCodeSnippet.node`?

Is the list any different from table, or does table have some special treatment?

Reply:
I'm not 100% what @patrickhulce is implying with this?  Nor can I see exactly what the format is?  Is it a details element that basically contains more details elements? like:
```javascript
{
   details: {
      items: [
         details,
         details,
         details...
         ],
}
```
?

Reply:
Yeah exactly, we have a `list` type that's very similar to `table`. In this case the `list` contains a `code-snippet` which contains a `node`. Not sure why this might cause problems.

Reply:
This isn't a blocker from a proto perspective, it will run fine.  Buuuuuut this level of nesting is _uncomfortable_ since our protos + the struct syntax are already highly nested.  How deep will this go?  Is there an upper limit?  Does it nest more than once?  

This nesting in the details (here and in tables) from a proto perspective is very bad and should be represented more flatly and predictably, but it is not a blocker and since we already do it for tables I think this is _okay_ as long as the nesting is not overly deep.

Reply:
Although if the list contains `code-snippet` objects we should call that out in the types instead of just throwing in more details objects as a proxy.

Reply:
Yeah that's sensible @mattzeunert!

I'm not entirely convinced we need a full-blown generic list details that people accidentally start using when we don't really want them to :)

I think I'd be a bit more comfortable with a code-snippet-list details type that doesn't do the nesting, but I don't feel *too* strongly.

Reply:
I think the main reason for the nesting is that we might want to put a code snippet inside a table at some point.

But we don't have any concrete plans and I'm not sure how we'd make that look good anyway, so I think having a `code-snippet-list`/`content-snippet-list`/... is fine.

Reply:
Split back into `snippet` and `list` since @brendankenny found a few of other places where we'd want to use a list.

Reply:
:( I'm sorry @mattzeunert! mind linking to the convo I missed?

Reply:
https://github.com/GoogleChrome/lighthouse/pull/6999#issuecomment-461657358